### PR TITLE
miner: fix regression, add test for starting while download

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**
fixes a regression introduced in https://github.com/ethereum/go-ethereum/pull/21536

### CODE REVIEW

File: miner/miner.go [hunk 1]
```diff
@@ -98,20 +99,25 @@ func (miner *Miner) update() {
...
 		case addr := <-miner.startCh:
-			miner.SetEtherbase(addr)
-			miner.worker.start()
+			if canStart {
+				miner.SetEtherbase(addr)
+				miner.worker.start()
+			}
+			shouldStart = true
 		case <-miner.stopCh:
 			miner.worker.stop()
```
Comment (holiman) [inline miner/miner.go:122]: Here, you need to set `shouldStart` to false, no?

Comment (MariusVanDerWijden) [reply miner/miner.go]: yes you're right!

**Review by holiman — APPROVED** [review]
LGTM

File: miner/miner.go [hunk 2]
```diff
@@ -98,21 +99,27 @@ func (miner *Miner) update() {
...
 					miner.worker.start()
 				}
 			}
 		case addr := <-miner.startCh:
-			miner.SetEtherbase(addr)
-			miner.worker.start()
+			if canStart {
+				miner.SetEtherbase(addr)
+				miner.worker.start()
+			}
```
Comment (holiman) [inline miner/miner.go:119]: Just a thought - what happens if user calls `miner.start(xx)` twice in a row? I'm assuming that doesn't cause any problems...?

Comment (MariusVanDerWijden) [reply miner/miner.go:119]: No, it doesn't cause any problems, just checked. Should I add a test for it?

**Comment by meowsbits** [comment]
What was the regression?

**Comment by holiman** [comment]
See https://github.com/ethereum/go-ethereum/pull/21536#issuecomment-690584213

**Pull request merged**